### PR TITLE
add a utility method for sorting CID slices

### LINKED ISSUE #46: It would be nice to add Less()
It's common to want to sort `Cid`, which requires an implementation of `Less()`. This can be done more efficiently inside `Cid` than outside because it won't require allocation if done inside (can directly access member fields which aren't exposed outside the package).

### PR DESCRIPTION
**Pull request by Stebalien**
blocked on the CID string refactor

fixes https://github.com/ipfs/go-cid/issues/46

### CODE REVIEW

**Review by dignifiedquire — COMMENTED** [review]
very cool, thank you

**Review by kevina — APPROVED** [review]
LGTM.

**Comment by kevina** [comment]
@Stebalien merge when ready.

File: slice.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+func (s Slice) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func (s Slice) Sort() {
+	sort.Sort(s)
+}
+
+// Sort sorts a slice of CIDs
+func Sort(s []cid.Cid) {
```
Comment (lanzafame) [inline slice.go:28]: Why not wrap either [`sort.Slice`](https://golang.org/pkg/sort/#Slice) or [`sort.SliceStable`](https://golang.org/pkg/sort/#SliceStable) and then the `Slice` type isn't required?

Comment (Stebalien) [reply slice.go:28]: `sort.Slice` uses reflection so it's not quite as fast. It may not make a difference but it's not exactly a ton of code.

Comment (lanzafame) [reply slice.go:28]: True, this may just be me having an aversion to named slice types more than anything else :)

Comment (Stebalien) [reply slice.go:28]: Well... when go introduces proper generics all this stuff will magically disappear. Until then, we have hand-coded monomorphization.

:disappointed:

**Pull request merged**